socorro/external/postgresql/connection_context.py

```diff
@@ -68,21 +81,41 @@ def __init__(self, config, local_config=None):
             local_config - this is the namespace within the complete config
                            where the actual database parameters are found"""
         super(ConnectionContext, self).__init__()
+
         self.config = config
+
         if local_config is None:
             local_config = config
+
+        if local_config['database_url']:
+            self.database_url = local_config['database_url']
+            url = urlparse.urlparse(local_config['database_url'])
+            local_config['database_username'] = url.username
+            local_config['database_password'] = url.password
+            local_config['database_hostname'] = url.hostname
+            local_config['database_port'] = url.port
+            local_config['database_name'] = url.path[1:]
+
+            self.db_kwargs = {
+                'host': url.hostname or 'localhost',
+                'user': url.username or os.environ.get('USER', 'test'),
```

Comment:
That makes sense to change around the variable setting.

As to your other extremely valid point, I don't have enough time left to debug the default settings as you suggest. 

Our options are: 

* abandon this patch based on your concern and I can come back to it around April 25th. The problem with that is that then the core socorro stuff doesn't support the Heroku URLs, and someone will need to take the Heroku URL from consul, parse it into our configs and update our existing configs as-is.
* Accept the patch even with the annoying defaults in it and revisit the problems with our unit tests, integration tests and jenkins/travis runs later